lib/matplotlib/pyplot.py

```diff
@@ -85,7 +85,8 @@
     import datetime
     import pathlib
     import os
-    from typing import Any, BinaryIO, Literal
+    from typing import Any, BinaryIO, Literal, TypeVar
+    from typing_extensions import ParamSpec
```

Comment:
I had been avoiding `typing_extensions` to avoid having a direct dependency on it.

That said, this _is_ guarded by an `if TYPE_CHECKING` so it is not a _runtime_ dependency, only a typechecking/dev dependency, (and it is a dependency of mypy, though I think I'd still prefer to be explicit if we are using it directly)

I had kind of written off ParamSpec as "interesting, but ultimately doesn't cover most of our actual needs" because most of our things that modify signatures do so with kw-only args, and that is explicitly not handled by paramspec. That said, I think it is perhaps defensible to say that saying it returns something like the input is better than just saying "any callable".

Reply:
I think adding `typing_extensions` to at least `requirements/testing/mypy.txt` would be prudent, just to ensure that _if_ it somehow becomes not transitive dependency we still pick it up for type checking (which I doubt, but explicitness is good)

Reply:
Added it to the requirements.